web/modules/custom/joinup_front_page/src/Plugin/Block/ExploreItemBlock.php

```diff
@@ -0,0 +1,203 @@
+   *   initialize the defined contexts by setting it to an array of context
+   *   values keyed by context names.
+   * @param string $plugin_id
+   *   The plugin_id for the plugin instance.
+   * @param mixed $plugin_definition
+   *   The plugin implementation definition.
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_manager
+   *   The entity manager.
+   * @param \Drupal\Core\Messenger\MessengerInterface $messenger
+   *   The messenger.
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, EntityTypeManagerInterface $entity_manager, MessengerInterface $messenger) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+    $this->entityManager = $entity_manager;
+    $this->messenger = $messenger;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('entity_type.manager'),
+      $container->get('messenger')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function build(): array {
+    $data = [
+      'solutions' => [
+        'name' => $this->t('Solutions'),
+        'data' => $this->getRdfEntity('solution'),
+        'url' => '/search?keys=&f[0]=type%3Asolution',
+      ],
+      'collections' => [
+        'name' => $this->t('Collections'),
+        'data' => $this->getRdfEntity('collection'),
+        'url' => '/search?keys=&f[0]=type%3Acollection',
+      ],
+      'news' => [
+        'name' => $this->t('News'),
+        'data' => $this->getContent('news'),
+        'url' => '/search?keys=&f[0]=type%3Anews',
+      ],
+      'events' => [
+        'name' => $this->t('Events'),
+        'data' => $this->getContent('events'),
+        'url' => '/search?keys=&f[0]=type%3Aevent',
+      ],
+    ];
+
+    return [
+      '#theme' => 'explore_block',
+      '#data' => $data,
+      '#cache' => [
+        'contexts' => $this->getCacheContexts(),
+        'tags' => $this->getCacheTags(),
+      ],
+    ];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getCacheContexts() {
+    return Cache::mergeContexts(parent::getCacheContexts(), ['route']);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getCacheTags() {
+    $tags = [
+      'node:news',
+      'node:events',
+      'rdf_entity:solution',
+      'rdf_entity:collection',
+    ];
+
+    return Cache::mergeTags(parent::getCacheTags(), $tags);
+  }
+
+  /**
+   * Get entity data.
+   *
+   * @param string $type
+   *   The type of content.
+   *
+   * @return array
+   *   List of content with view mode "explore_item".
+   *
+   * @throws \Drupal\Component\Plugin\Exception\InvalidPluginDefinitionException
+   * @throws \Drupal\Component\Plugin\Exception\PluginNotFoundException
+   */
+  private function getContent($type): array {
```

Comment:
In Joinup we are using the Drupal philosophy that every class should be reusable and extendable so we avoid the use of private methods. This can be a protected one. Also let's add the `string $type` type hint.